contracts/utils/CommitteeUpgradeable.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "../interfaces/IBridgeCommittee.sol";
+import "./MessageVerifier.sol";
+
+abstract contract CommitteeUpgradeable is
+    UUPSUpgradeable,
+    MessageVerifier,
+    ReentrancyGuardUpgradeable
+{
+    bool private _upgradeAuthorized;
+
+    function __CommitteeUpgradeable_init(address _committee) internal onlyInitializing {
+        __ReentrancyGuard_init();
+        __MessageVerifier_init(_committee);
+        committee = IBridgeCommittee(_committee);
+    }
+
+    function _authorizeUpgrade(address) internal view override {
+        require(_upgradeAuthorized, "SuiBridge: Unauthorized upgrade");
+    }
+
+    function upgradeWithSignatures(bytes[] memory signatures, BridgeMessage.Message memory message)
```

Comment:
I think this does not work - different contract needs to have separate nonces for upgrade, right?

Reply:
<img width="902" alt="Screenshot 2024-01-26 at 12 35 38 PM" src="https://github.com/MystenLabs/evm-sui-bridge/assets/24499532/0c161c3a-fe9b-41ff-82c8-1a3e09cf6988">